ajd/Animal/lib/Animal.pm

```diff
@@ -48,6 +71,70 @@ sub sound {
   die 'You must define sound() in a subclass';
 }
 
+=head2 default_colour
+
+The default colour of an animal. This must be overidden.
+
+=cut
+
+sub default_colour {
+  die 'You must define default_colour() in a subclass';
+}
+
+=head2 name
+
+Call with a name to set the name. Call with no params to get the name.
+
+=cut
+
+sub name {
+
+  my $self = shift;
+
+  if ( ref( $self ) ) {
+    if ( scalar( @_ ) ) {
+      $self->{name} = shift;
```

Comment:
As you're not returning anything anyway, might as well return `$self` for that nice property chaining behaviour:
```perl
$horsey->name( 'Fancy dancy magic prancy' )->colour( 'Rainbow' );
```